adapter/service/outlvlswitch/service.go

```diff
@@ -0,0 +1,184 @@
+package outlvlswitch
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/adapter/cache"
+	"github.com/futurehomeno/fimpgo"
+	"github.com/futurehomeno/fimpgo/fimptype"
+)
+
+// Constants defining important properties specific for the service.
+const (
+	MaxLvl = "max_lvl"
+	MinLvl = "min_lvl"
+	SwType = "sw_type" // "on_off" or "up_down"
+)
+
+// DefaultReportingStrategy is the default reporting strategy used by the service for periodic reports.
+var DefaultReportingStrategy = cache.ReportAtLeastEvery(30 * time.Minute)
+
+// Controller is an interface representing an actual device.
+// In a polling scenario implementation might require some safeguards against excessive polling.
+type Controller interface {
+	// LevelReport returns a current level value.
+	LevelReport() (float64, error)
```

Comment:
Controller naming should follow how other controllers were implemented. The idea is to not have naming conflicts and each method is clearly associated with a service. So I suggest more like:

LevelSwitch(Level)Report
LevelSwitchBinaryReport
SetLevelSwitchLevel
SetLevelSwitchBinaryState

These are just from top of my head.